controllers/restore_controller.go

```diff
@@ -800,3 +810,124 @@ func processRestoreWait(
 
 	return false, ""
 }
+
+const (
+	// import-controller-config ConfigMap name and namespace
+	importControllerConfigName = "import-controller-config"
+	importControllerNamespace  = "multicluster-engine"
+	// autoImportStrategy key in the ConfigMap
+	autoImportStrategyKey = "autoImportStrategy"
+	// ImportOnly is the default strategy for new ACM 2.14+ hubs
+	importOnlyStrategy    = "ImportOnly"
+	importAndSyncStrategy = "ImportAndSync"
+	// immediate-import annotation key
+	immediateImportAnnotation = "import.open-cluster-management.io/immediate-import"
+)
+
+// isImportOnlyStrategySet checks if the import strategy is set to ImportOnly
+// Returns true if:
+// - ConfigMap doesn't exist (ImportOnly is the default for new ACM 2.14+ hubs)
+// - ConfigMap exists but autoImportStrategy is not set (defaults to ImportOnly)
+// - ConfigMap exists and autoImportStrategy is explicitly set to ImportOnly
+func isImportOnlyStrategySet(ctx context.Context, c client.Client) bool {
+	logger := log.FromContext(ctx)
+
+	configMap := &v1.ConfigMap{}
+	err := c.Get(ctx, types.NamespacedName{
+		Name:      importControllerConfigName,
+		Namespace: importControllerNamespace,
+	}, configMap)
+
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			// ConfigMap doesn't exist, ImportOnly is the default
+			logger.Info("import-controller-config ConfigMap not found, ImportOnly strategy is default")
+			return true
+		}
+		// Error getting ConfigMap, assume ImportOnly for safety
+		logger.Error(err, "Error getting import-controller-config ConfigMap, assuming ImportOnly")
+		return true
+	}
+
+	// ConfigMap exists, check the autoImportStrategy value
+	strategy, exists := configMap.Data[autoImportStrategyKey]
+	if !exists || strategy == "" {
+		// autoImportStrategy not set, ImportOnly is the default
+		logger.Info("autoImportStrategy not set in ConfigMap, ImportOnly strategy is default")
+		return true
+	}
+
+	// Check if strategy is ImportOnly
+	isImportOnly := strings.EqualFold(strings.TrimSpace(strategy), importOnlyStrategy)
+	logger.Info("autoImportStrategy found", "strategy", strategy, "isImportOnly", isImportOnly)
+	return isImportOnly
+}
+
+// shouldSkipManagedClusterForAnnotation checks if a ManagedCluster should be skipped
+// when adding the immediate-import annotation
+func shouldSkipManagedClusterForAnnotation(mc *clusterv1.ManagedCluster, localClusterName string) bool {
+	// Skip local-cluster by label
+	if isLocalCluster(mc) {
+		return true
+	}
+	// Skip if name matches local cluster name
+	if localClusterName != "" && mc.Name == localClusterName {
```

Comment:
is this necessary ? isLocalCluster should find the local cluster using the annotation